File: src/main/java/nl/crosscode/x509certbuddy/actions/ExportPEMAction.java
package nl.crosscode.x509certbuddy.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import nl.crosscode.x509certbuddy.ui.Exporters;
import nl.crosscode.x509certbuddy.ui.x509CertAssistant;
import nl.crosscode.x509certbuddy.x509CertAssistantFactory;
import org.jetbrains.annotations.NotNull;

public class ExportPEMAction extends AnAction {
    private final Exporters exporters;

    public ExportPEMAction(Exporters exporters) {
        this.exporters = exporters;
        this.getTemplatePresentation().setText("Export PEM");
    }

    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        exporters.exportPEM(null);
    }

    @Override
    public void update(@NotNull AnActionEvent e) {
        x509CertAssistant assistant = x509CertAssistantFactory.getInstance(e.getProject());
        e.getPresentation().setEnabled(assistant!=null?assistant.hasCertSelected():false);
        super.update(e);
    }
}

File: src/main/kotlin/nl/crosscode/x509certbuddy/actions/ExportPEMAction.kt
package nl.crosscode.x509certbuddy.actions

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import nl.crosscode.x509certbuddy.X509CertAssistantFactory
import nl.crosscode.x509certbuddy.ui.Exporters

class ExportPEMAction(private val exporters: Exporters) : AnAction() {
    init {
        this.templatePresentation.text = "Export PEM"
    }

    override fun actionPerformed(e: AnActionEvent) {
        exporters.exportPEM()
    }

    override fun update(e: AnActionEvent) {
        if (e.project == null) {
            e.presentation.isEnabled = false
            return
        }
        val assistant = X509CertAssistantFactory.getInstance(e.project!!)
        e.presentation.isEnabled = assistant?.hasCertSelected() ?: false
        super.update(e)
    }
}
